Add examples for various inventory setups to the documentation (#62323)
ansible/ansible: docs/docsite/rst/user_guide/intro_inventory.rst

Inventory setup examples
========================

.. _inventory_setup-per_environment:

Example: One inventory per environment
--------------------------------------

If you need to manage multiple environments it's sometimes prudent to
have only hosts of a single environment defined per inventory. This
way, it is harder to, for instance, accidentally change the state of
nodes inside the "test" environment when you actually wanted to update
some "staging" servers.

For the example mentioned above you could have an
:file:`inventory_test` file:

.. code-block:: ini

  [dbservers]
  db01.test.example.com
  db02.test.example.com

  [appservers]
  app01.test.example.com
  app02.test.example.com
  app03.test.example.com

That file only includes hosts that are part of the "test"
environment. Define the "staging" machines in another file
called :file:`inventory_staging`:

.. code-block:: ini

  [dbservers]
  db01.staging.example.com
  db02.staging.example.com

  [appservers]
  app01.staging.example.com
  app02.staging.example.com
  app03.staging.example.com

To apply a playbook called :file:`site.yml`
to all the app servers in the test environment, use the
following command::

>   ansible-playbook -i inventory_test site.yml -l app-servers

Reviewer: ```suggestion
  ansible-playbook -i inventory_test site.yml -l appservers
```

Author: 🙈😂

.. note:: If you're reading the docs from the beginning, this may be the first example you've seen of an Ansible playbook. This is not an inventory file.
          Playbooks will be covered in great detail later in the docs.

.. _inventory_setup_examples:

Inventory setup examples
========================

.. _inventory_setup-per_environment:

Example: One inventory per environment
--------------------------------------

If you need to manage multiple environments it's sometimes prudent to
have only hosts of a single environment defined per inventory. This
way, it is harder to, for instance, accidentally change the state of
nodes inside the "test" environment when you actually wanted to update
some "staging" servers.

For the example mentioned above you could have an
:file:`inventory_test` file::

>   [database-servers]

Reviewer 2: avoid dashes, -servers seems redundant anyways

Author: Fixed :) The "servers" postfix is consistent with other group-names in that document, though, so I opted to keep it.

  db01.test.example.com
  db02.test.example.com

  [app-servers]
  app01.test.example.com
  app02.test.example.com
  app03.test.example.com

> In there are only host that are part of the "test"

Reviewer: ```suggestion
That file only includes hosts that are part of the "test"
```

> environment. "Staging" machines would be defined in another file

Reviewer: ```suggestion
environment. Define the "staging" machines in another file
```

called :file:`inventory_staging`::

  [database-servers]
  db01.staging.example.com
  db02.staging.example.com

  [app-servers]
  app01.staging.example.com
  app02.staging.example.com
  app03.staging.example.com

> If you then have a playbook called :file:`site.yml`, then you'd apply

Reviewer: ```suggestion
To apply a playbook called :file:`site.yml`
```

> that to all the app-servers in the test environment using the

Reviewer: ```suggestion
to all the app-servers in the test environment, use the
```

following command::

  ansible-playbook -i inventory_test site.yml -l app-servers

.. _inventory_setup-per_function:

Example: Group by function
--------------------------

In the previous section you already saw an example for using groups in
order to cluster hosts that have the same function. This allows you,
for instance, to define firewall rules inside a playbook or role

> without affecting database servers::

Reviewer: please use code-block here

Author: Fixed :)
